app.rb

```diff
@@ -367,35 +373,31 @@ def get_github_repos
   #
 
   get "/" do
-    @github_repos = get_github_repos
-
     haml :index
   end
 
-    Nailed::Config.products.each do |product|
-      get "/#{product.tr(" ", "_")}/bugzilla" do
-        @github_repos = get_github_repos
-
-        @product = get_label(product)
-        @product_ = product.tr(" ", "_")
+  Nailed::Config.products.each do |product|
+    get "/#{product.tr(" ", "_")}/bugzilla" do
+      @product = get_label(product)
+      @product_ = product.tr(" ", "_")
 
-        haml :bugzilla
-      end
+      haml :bugzilla
     end
+  end
 
-  github_repos = Pullrequest.order(Sequel.desc(:created_at)).all.map do |row|
-    [row.oname, row.rname]
-  end.uniq
-
-  github_repos.each do |repo|
-    get "/github/#{repo[0]}/#{repo[1]}" do
-      @github_repos = get_github_repos
+  Nailed::Config.supported_vcs.each do |vcs|
+    changes_repos = Changerequest.where(origin: vcs).order(Sequel.desc(:created_at)).all.map do |row|
+      [row.oname, row.rname, row.url.rpartition("/").first]
+    end.uniq
 
-      @repo = repo[1]
-      @org = repo[0]
-      @github_url_all_pulls = "https://github.com/#{@org}/#{repo}/pulls"
+    changes_repos.each do |repo|
```

Comment:
Since you generated `changes_repo` from the database and not from the config file, you will only generate routes for repos with open MR's/PR's, which means, repo's without those will cause Sinatra errors. See proposed [WIP] fix here: https://github.com/parlt91/nailed/pull/1/files